src/ray/gcs/gcs_server/gcs_node_manager.cc

```diff
@@ -48,7 +48,12 @@ void GcsNodeManager::NodeFailureDetector::HandleHeartbeat(
   }
 
   iter->second = num_heartbeats_timeout_;
-  heartbeat_buffer_[node_id] = heartbeat_data;
+  if (heartbeat_data.should_global_gc() ||
+      heartbeat_data.resources_available_label_size() > 0 ||
+      heartbeat_data.resources_total_label_size() > 0 ||
+      heartbeat_data.resource_load_label_size() > 0) {
+    heartbeat_buffer_[node_id] = heartbeat_data;
```

Comment:
It means we still broadcast all heartbeat back to raylet right (although there's change)? Is it going to be covered by the next PR, or you decided not to cover this case?

Reply:
It broadcast all heartbeat which has changed `should_global_gc` flag or any resource.  If heartbeat of one node doesn't change any all, it would not be broadcasted.
Also, if the heartbeat only contains flag or load/available/total resource, then only the changed part would be included in the heartbeat batch message.